### PR TITLE
Update documentation on qudits

### PR DESCRIPTION
**Pull request by dabacon**

### CODE REVIEW

File: docs/qudits.ipynb [hunk 1]
```diff
@@ -84,16 +84,14 @@
     "id": "9c07f9b01c71"
```
[the remark this answers is not in the document]
Comment (dabacon) [reply docs/qudits.ipynb:84]: yeah, think also that we decided that `cirq.Class` could be used for first instance and then not repeated. For notebooks this doesn't seem to matter as it doesn't deep link.

[the remark this answers is not in the document]
Comment (dabacon) [reply docs/qudits.ipynb:84]: fixed

**Comment by dabacon** [comment]
Removed outputs from notebook.

**Comment by mpharrigan** [comment]
I just took a look at the beginning of this before realizing how extensive the changes were. I'm OOO until monday, so unassigning myself for review

File: docs/qudits.ipynb [hunk 2]
```diff
@@ -102,34 +100,45 @@
...
+    "class QutritPlusGate(cirq.Gate):\n",
+    "    \"\"\"A gate that adds one in the computational basis of a qutrit.\n",
+    "    \n",
+    "    This gate acts on three-level systems. In the computational basis of\n",
+    "    this system it enacts the transformation U|x> = |x + 1 mod 3>, or\n",
+    "    in other words U|0> = |1>, U|1> = |2>, and U|2> = |0>.\n",
+    "    \"\"\"\n",
+    "    \n",
     "    def _qid_shape_(self):\n",
+    "        # By implementing this method this gate implements the\n",
```
Comment (dstrain115) [inline docs/qudits.ipynb:117]: Optional:  Can this be a docstring with triple quotes rather than line comments?  Same below.

**Review by dstrain115 — APPROVED** [review]
LGTM with a few nits.

File: docs/qudits.ipynb [hunk 3]
```diff
@@ -84,16 +84,14 @@
...
     "\n",
-    "Qudits with known values for d have specific names. A qubit has dimension 2, a qutrit has dimension 3, a ququart has dimension 4, and so on.\n",
-    "In Cirq, qudits work exactly like qubits except they have a `dimension` attribute other than 2, and they can only be used with gates specific to that dimension.\n",
+    "Qudits with known values for d have specific names. A **qubit** has dimension 2, a **qutrit** has dimension 3, a **ququart** has dimension 4, and so on.\n",
+    "In Cirq, qudits work exactly like qubits except they have a `dimension` attribute different than 2, and they can only be used with gates specific to that dimension. In cirq, both qubits and qudits are subclasses of the class `cirq.Qid`. \n",
     "\n",
-    "Both qubits and qudits are represented by a `Qid` object.\n",
+    "To apply a gate to some qudits, the dimensions of the qudits must match the dimensions it works on.  For example, if a gate represents a unitary evolution on three qudits, a qubit, a qutrit, and another qutrit, the gate's \"qid shape\" is `(2, 3, 3)` and its `on` method will accept exactly 3 `Qid`s with dimension 2, 3, and 3, respectively.\n",
```
[the remark this answers is not in the document]
Comment (dabacon) [reply docs/qudits.ipynb:92]: rewrote as three sentences to make clearer.

File: docs/qudits.ipynb [hunk 4]
```diff
@@ -84,16 +84,14 @@
     "id": "9c07f9b01c71"
    },
    "source": [
-    "Most of the time in quantum computation, we work with qubits, which are 2-level quantum systems. A qu-*d*-it is a generalization of a qubit to a d-level or d-dimension system.\n",
+    "Most of the time in quantum computation, we work with qubits, which are 2-level quantum systems. But it is possible to also define quantum computation with higher dimensional systems. A qu-*d*-it is a generalization of a qubit to a d-level or d-dimension system.  For example, the state of a single qubit is a superposition of two basis states, $|\\psi\\rangle=\\alpha|0\\rangle+\\beta|1\\rangle$, whereas the state of a qudit for a three dimensional system is a superposition of three basis states $|\\psi\\rangle=\\alpha|0\\rangle+\\beta|1\\rangle+\\gamma|2\\rangle$.\n",
```
[the remark this answers is not in the document]
Comment (dabacon) [reply docs/qudits.ipynb:87]: done

**Pull request merged**